robustness_metrics/metrics/uncertainty.py

```diff
@@ -787,13 +802,34 @@ def update_state(self, y_true: Sequence[float], y_pred: Sequence[float],
         `tf.math.reduce_max(logits)`. Shape (batch_size, ).
       confidence: The confidence score where higher value indicates lower
         uncertainty. Values should be within [0, 1].
+      sample_weight: Optional weighting of each example.
+        Defaults to 1. Shape(batch_size, ).
       **kwargs: Additional keyword arguments.
     """
     # Creates binary 'label' of prediction correctness, shape (batch_size, ).
     scores = tf.convert_to_tensor(confidence, dtype=self.dtype)
+    y_true = tf.convert_to_tensor(y_true)
+    y_pred = tf.convert_to_tensor(y_pred)
+
+    assert tf.rank(y_pred) == tf.rank(y_pred) == tf.rank(confidence)
+
+    # Flatten all to [N, ].
+    if tf.rank(y_true) != 1:
+      y_true = tf.reshape(y_true, [-1])
+      scores = tf.reshape(scores, [-1])
+      y_pred = tf.reshape(y_pred, [-1])
+
     labels = _compute_correct_predictions(
         y_true, y_pred, dtype=self.dtype)
 
+    # use sample weight if provided
+    if sample_weight is not None:
+      sample_weight = tf.convert_to_tensor(sample_weight)
+      if tf.rank(sample_weight) != 1:
+        sample_weight = tf.reshape(sample_weight, [-1])
+      scores = tf.boolean_mask(scores, sample_weight)
+      labels = tf.boolean_mask(labels, sample_weight)
```

Comment:
Same comment as in https://github.com/google-research/robustness_metrics/pull/77: if we add a `sample_weight` to the API here, we'd need to follow the tf.keras.metrics.Metric API and have `sample_weight` be an arbitrary _continuous_ tensor. Binary masking is a subset of possible case, and we'd need to think more about how to support that general case here. For binary masking, I'd suggest moving that to the calling site for now.